sympy/vector/tests/test_field_functions.py

```diff
@@ -11,70 +11,73 @@
                                     scalar_potential_difference)
 from sympy.utilities.pytest import raises
 
-C = CoordSysCartesian('C')
-i, j, k = C.base_vectors()
-x, y, z = C.base_scalars()
-delop = C.delop
+cartesian = CoordSystem3D('C')
+spherical = CoordSystem3D('spherical', DifferentialClass.build_instance_from_lambda(
```

Comment:
Yes, this is just a draft. For *cylindrical* and *spherical* coord systems we could insert the three metric parameters directly (i.e. not deriving them). For custom coordinate system, leave this derivation function to automatically determine the differential operators.